### DISCUSSION TITLE
Is it can be implemented inside a nextJS project?

### DISCUSSION THREAD

**Discussion by aviramroi** [TanStack/router#95] [Q&A] [answered]
I have a nextJS project that I want to implement inside a module with hashRouting, can I do this module with react-location ?
tnx

**Comment by tannerlinsley** [comment, marked as answer]
Yes. That’s possible.

**Comment by essential-randomness** [comment]
Piggybacking on this question, I'm currently using [query-string](https://github.com/sindresorhus/query-string) in a NextJS project to do search params routing. I'm not very satisfied with the experience, and I'd like to give this library a try since I've been loving the experience of using react-query.

I'd assume switching the whole Router experience from the NextJS baked-in router to react-location would end up messy (at least I assume), but I'd be interested in doing so just for search params. Would you consider "worth it" to do so, or is this library overkill for just this use case? Would adding react-location provide any other benefit I'm not seeing at a glance in a NextJS project?

Thank you for help! And thank you for creating a library that does treat search params as the important routing tool they are.
